packages/blade/src/components/TopNav/_decisions/decisions.md

```diff
@@ -122,6 +122,157 @@ The container for the right-aligned actions like search, buttons, and user profi
 
 ### TabNav
 
+The horizontal navigation bar that contains multiple navigation items, each represented by TabNavItem. 
+
+TabNav's interaction is as follows: 
+
+1. There might already be items under More even at the highest screen size. These items might need a custom order.
+2. Once the screen size starts getting reduced, more items will start moving into More.
+3. The order of these items would follow a logical order. They keep getting added ahead of the already existing items. So the last item to go into More is the first item that you see when you click on More.
+
+TabNav automatically handles all these edge cases & responsiveness.
```

Comment:
add 1 liner that talks about that we are using function as children pattern to achieve all these edge cases and probably link to this https://reactpatterns.js.org/docs/function-as-child-component

Reply:
added